doc/md/guides/deploying/flyio.mdx

```diff
@@ -0,0 +1,167 @@
+---
+id: fly.io
+slug: /guides/deploying/fly.io
+title: Using Atlas with Fly.io
+---
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+[Fly.io](https://fly.io/) is a platform for running full stack apps and databases close to the users. Under the hood, Fly
+converts Docker images (OCI images) to [Firecracker microVMs](https://firecracker-microvm.github.io/). Fly allows
+the deployment of the app in different regions and route the requests based on the [app load and user closeness](https://fly.io/docs/reference/load-balancing/).
+
+Apps on Fly can be [deployed in one of three ways](https://fly.io/docs/reference/builders/): using a Dockerfile, a
+Docker image or a [buildpack](https://buildpacks.io/).
+
+In this guide, we will demonstrate how Atlas can be used to perform database schema migrations for Fly.io deployments process using a Dockerfile.
+We will assume that you have already have a Fly project and are able to deploy it, if you are new to Fly, [check the getting started guide](https://fly.io/docs/speedrun/).
+
+## Release command
+
+When you configure your Fly project, you can define a _release command_. This command is executed during the release phase, before the new version of your application is deployed.  If it fails, i.e exits with a status other than zero, Fly marks the deployment as failed mas well. The release command is the
+[recommended way](https://fly.io/docs/reference/configuration/#the-deploy-section) to run database migrations on Fly.
+
+:::note
+The release command only allows executing commands that are present in the application image, meaning that we
+have to embed the Atlas binary with our application. [We usually suggest a separate step for handling the
+migration](https://atlasgo.io/guides/deploying/intro#running-migrations-as-part-of-deployment-pipelines),
+but since this is a requirement from Fly, we don't have an option.
+:::
+
+## Defining the Dockerfile
+
+Using Docker multi-stage builds, we can compose lightweight images from multiple steps that use heavier base images. In this guide, we will use a Go app as an example so we can use a step for building the target binary and a separate one for producing the runtime container, that does not need to include the build tools at all.
```

Comment:
```suggestion
Using Docker multi-stage builds, we can compose lightweight images from multiple steps that use heavier base images. In this guide, we will use a Go app as an example. Because Go is a compiled language, so we can use a separate step for building the target binary and another for producing the runtime container. This way the runtime environment can be smaller, omitting the build environment.
```